File: 0260-single-number-iii/0260-single-number-iii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        n = (w:=reduce(lambda x,y:x^y, nums,0))&-w
        print(n)
        ans=reduce(lambda x,y : [x[0]^y if n&y==0 else x[0], x[1]^y if n&y else x[1]] ,nums,[0,0])
        return ans

File: 0260-single-number-iii/0260-single-number-iii.java
class Solution {
    public int[] singleNumber(int[] nums) {
        int xors = 0;
        for(int n:nums){
            xors^=n;
        }
        int bit=xors&(-xors);
        int[] ans={0,0};
        for(int n:nums){
            if((bit&n)==0){
                ans[0]^=n;
            }else{
                ans[1]^=n;
            }
        }
        return ans;
    }
}